[CS2113-W13-1] DoctorDuke

Jeraldchen opened the pull request:
Doctor Duke is a desktop app for diagnosing patients who have mild illnesses, optimized for use via a Command Line Interface (CLI). If you can type fast, Doctor Duke can diagnose illnesses and dispense medicine accordingly faster than visiting a General Practitioner.

Jeraldchen committed:
Add and update UserGuide for v1.0

Jeraldchen committed:
Merge pull request #59 from Jeraldchen/Jeraldchen-UserGuide

Jeraldchen user guide

Jeraldchen committed:
Add logging for error handling in storage and information classes

Jeraldchen committed:
Remove * from import to pass style checks

Jeraldchen committed:
Chang logger code line to pass check style

Jeraldchen committed:
Merge pull request #60 from Jeraldchen/Jeraldchen-Logging

Jeraldchen logging

Brennanzuz committed:
Added Javadoc to diagnosis package classes and methods.

Added documentation to important methods and classes involving diagnosis.
Changed author comments to Javadoc @Author tags.
Remove commented test code for diagnosis again.
Removed some redundancies.

Brennanzuz committed:
Added assertions and logging.

Added an assertion in Diagnosis.java to prevent a divide by zero in the event the programmer created an Illness child with no symptoms.
Added logging to signal when diagnosis was ran, and what is the comparison value to each of the possible illnesses.
Fixed a typo on authorship @Author tag.

Brennanzuz committed:
Fixed failed checkstyleTest lines.

Brennanzuz committed:
Merge pull request #61 from Brennanzuz/branch-brannon-202303071516

Branch brannon 202303071516

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Enable assertions in build.gradle

Jeraldchen committed:
Merge pull request #64 from Jeraldchen/master

Enable assertion in build.gradle

Jeraldchen committed:
Enable multiple symptoms input and fix headache symptom bug

Jeraldchen committed:
Enable multiple symptoms input feature

Jeraldchen committed:
Merge pull request #65 from Jeraldchen/Jeraldchen-MultipleInputs

Jeraldchen multiple inputs

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Thunderdragon221 committed:
Prevent duplicate passwords and hash passwords

Thunderdragon221 committed:
Merge pull request #67 from Thunderdragon221/branch-passwordSecurity

Prevent duplicate passwords and hash passwords

Jeraldchen committed:
no message

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Commit:
Allow user to see medicine prescription history.
Added more medications for illnesses.

tanyizhe committed:
Merge pull request #68 from tanyizhe/tanyizhe-medicineHistory

Allow user to see medicine prescription history.

Geeeetyx committed:
Merge branch 'AY2223S2-CS2113-W13-1:master' into master

Geeeetyx committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Resolve merge conflicts in data.txt

Commit:
Added more medicines

tanyizhe committed:
Merge pull request #71 from tanyizhe/master

Added more medicines

Brennanzuz committed:
Added more symptoms the users can input.

Added 13 more options for the user to input when entering their symptoms.
Added a new function addSymptom to add users' symptoms to the symptoms ArrayList.
Added error handling in Patient.java in the event no medication is available for the illness.
Removed boilerplate lines in case statement when user inputs their symptoms and moved them to addSymptom
Fixed Insomnia.java to display the correct symptom.

Brennanzuz committed:
Fixed order of @author tags in Javadoc

Fixed order of @author tags in Javadoc to pass Gradlew checks.

Brennanzuz committed:
Merge pull request #73 from Brennanzuz/branch-brannon-202303071516

Branch brannon 202303071516

Geeeetyx committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Geeeetyx committed:
feat: update user guide, improve UI

Geeeetyx committed:
fix: added assertions

Geeeetyx committed:
fix: fix code style

Geeeetyx committed:
fix: fix test-ui-test

Geeeetyx committed:
fix: fix test-ui-test

Geeeetyx committed:
Merge pull request #78 from Geeeetyx/Geeeetyx_3

Update user guide

Brennanzuz committed:
Added diagrams folder.

Added diagrams folder with a single .puml file named in PascalCase to standardise.

Brennanzuz committed:
Merge pull request #79 from Brennanzuz/branch-brannon-202303071516

Added diagrams folder.

Brennanzuz committed:
Added documentation and class diagram for DeveloperGuide.md

Added DiagnosisClassDiagram.puml UML class diagram to document the structure of the diagnosis package classes.
Added documentation in DeveloperGuide.md to describe how diagnosis returns possible illnesses given the patient's symptoms.
Added Style.puml for some easy referencing of tertiary colours.
Removed 2 duplicate illnesses created in `ALL_ILLNESSES`

Brennanzuz committed:
Merge pull request #80 from Brennanzuz/branch-brannon-202303071516

Added documentation and class diagram for DeveloperGuide.md

Commit:
Fixed bug regarding MedicineManager.
Added implementation of MedicineManager and Medicine classes to Developer guide.
Medicine history is now stored.

Commit:
Fixed bug regarding MedicineManager.
Added implementation of MedicineManager and Medicine classes to Developer guide.
Medicine history is now stored.

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/DeveloperGuide.md
#	src/main/java/seedu/duke/diagnosis/Diagnosis.java

tanyizhe committed:
Merge pull request #81 from tanyizhe/master

Improved features and fixed bugs with medicine classes

Geeeetyx committed:
Merge branch 'AY2223S2-CS2113-W13-1:master' into master

Geeeetyx committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Thunderdragon221 committed:
Add Storage UML Class Diagram

Thunderdragon221 committed:
Merge pull request #83 from Thunderdragon221/branch-StorageUMLDiagram

Add Storage UML Class Diagram

Jeraldchen committed:
Fix merge conflicts

Geeeetyx committed:
Merge branch 'AY2223S2-CS2113-W13-1:master' into master

Geeeetyx committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Geeeetyx committed:
Add Patient UML Diagram

Geeeetyx committed:
Merge pull request #84 from Geeeetyx/Geeeetyx_DG

Patient UML Diagram

Geeeetyx committed:
Update DG

Geeeetyx committed:
Merge pull request #85 from Geeeetyx/Geeeetyx_DG

Update DG

Jeraldchen committed:
Update Class Diagrams for DG

Jeraldchen committed:
Merge pull request #86 from Jeraldchen/Jeraldchen-DeveloperGuideDraft

Jeraldchen developer guide draft

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Fix reposense authorship syntax

Jeraldchen committed:
Merge pull request #87 from Jeraldchen/master

Fix reposense authorship syntax

Jeraldchen committed:
Edit developer guide

Jeraldchen committed:
Merge pull request #88 from Jeraldchen/master

Edit developer guide

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Added more JUnit Tests

Jeraldchen committed:
Edit syntax to pass gradle check

Jeraldchen committed:
Edit naming of hypertension to pass CI check

Jeraldchen committed:
Edit line of code on 155 to newline to pass gradle check

Jeraldchen committed:
Merge pull request #91 from Jeraldchen/Jeraldchen-MoreJUnitTests

Jeraldchen more j unit tests

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Add try catch blocks and added new feature to view symptoms history

Jeraldchen committed:
Merge pull request #92 from Jeraldchen/Jeraldchen-trycatch

Jeraldchen trycatch

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Commit:
Added assert lines and more JUnit testing

tanyizhe committed:
Merge pull request #93 from tanyizhe/master

Added assert lines and more JUnit testing

Geeeetyx committed:
Merge branch 'AY2223S2-CS2113-W13-1:master' into master

Geeeetyx committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Geeeetyx committed:
Update DG for Patient Class

Geeeetyx committed:
Merge pull request #94 from Geeeetyx/Geeeetyx_DG2

Update DG for patient class

Geeeetyx committed:
update JUnit test for patient class

Geeeetyx committed:
update text-test

Geeeetyx committed:
Merge pull request #95 from Geeeetyx/Geeeetyx_DG2

update JUnit test for patient class

Commit:
Fixed bugs and added more medicine to diagnosis pairings.

Commit:
Merge branch 'tanyizhe-bugFix'

tanyizhe committed:
Merge pull request #96 from tanyizhe/master

Fixed bug with medicine file storage

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Add new delete symptom feature

Jeraldchen committed:
Merge pull request #98 from Jeraldchen/Jeraldchen-deleteSymptom

Jeraldchen delete symptom

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Updated user guide and fixed bug in delete symptom

Jeraldchen committed:
Merge pull request #100 from Jeraldchen/Jeraldchen-UserGuide2.0

Jeraldchen user guide2.0

Geeeetyx committed:
Merge branch 'AY2223S2-CS2113-W13-1:master' into master

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
Update UG and DG

Geeeetyx committed:
Merge pull request #101 from Geeeetyx/Geeeetyx_DG3

Update UG and DG

Commit:
Add find and list features for medicine.
Updated user guide.

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/duke/ui/Menu.java
#	src/main/java/seedu/duke/ui/Parser.java

tanyizhe committed:
Merge pull request #102 from tanyizhe/master

Add list and find features for medicine

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
Update UG and DG

Geeeetyx committed:
Merge pull request #103 from Geeeetyx/Geeeetyx_DG4

Update UG and DG

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Thunderdragon221 committed:
edit //@@author comments to reflect correctly

Thunderdragon221 committed:
Merge pull request #104 from Thunderdragon221/branch-edit_/@@author

edit //@@author comments to reflect correctly

Brennanzuz committed:
Added instructions to UserGuide.md

Added on some instructions to clarify to users about finishing the reporting of their symptoms.

Brennanzuz committed:
Merge pull request #105 from Brennanzuz/branch-brannon-202303071516

Added instructions to UserGuide.md

Thunderdragon221 committed:
add sequence diagrams for Storage class

Thunderdragon221 committed:
Merge pull request #106 from Thunderdragon221/branch-addSequenceDiagrams

add sequence diagrams for Storage class

Brennanzuz committed:
Added DiagnosisSequenceDiagram.puml

Added the UML sequence diagram to describe the diagnosis algorithm in Diagnosis.java
Added DiagnosisSequenceDiagram.png into the DeveloperGuide.md.

Brennanzuz committed:
Merge pull request #107 from Brennanzuz/branch-brannon-202303071516

Added DiagnosisSequenceDiagram.puml

tanyizhe committed:
Create MedicineSequenceDiagram.puml

tanyizhe committed:
Add medicinesequencediagram

tanyizhe committed:
Update DeveloperGuide.md

tanyizhe committed:
Update DeveloperGuide.md

wuzhzn commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+[comment]: <> (//@@author Jeraldchen)
+
+### User Interface
+![UIClass.png](diagrams%2FUIClass.png)
```
Should the class diagram include the connection to different classes and use the correct syntax? Perhaps it should include the private and public accessibility in the class diagram

Arif-Khalid replied:
Are all the attributes and operations in Menu, Information and Parser class-level? Shouldn't only the class-level attributes and operations be underlined in a class diagram?

darrenlsx replied:
Overall very neat diagram for UserInterface

wuzhzn commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+
+#### Diagnosis algorithm
+
+![DiagnosisSequenceDiagram.png](diagrams%2FDiagnosisSequenceDiagram.png)
```
Should the loop at the top contain correct loop syntax? Perhaps it should follow the syntax of the loop below.

wuzhzn replied:
Perhaps the illness should not be labelled as a constructor

wuzhzn replied:
For the medicine Manager, perhaps the activation bar should be closed or labelled as closed

FanZixian replied:
Perhaps the syntax of loop is ambiguous.

Arif-Khalid replied:
Should the lifespan(dotted line) of the entities still be there after the entity is deleted?

Arif-Khalid replied:
Should MedicineManager and Diagnosis be labelled as only invoking their static members, (<<class>>), doesn't this conflict with the class diagram above?

darrenlsx replied:
loop syntax could be looked at again, X markers could also perhaps be placed at the more appropriate spots to show exactly when the lifespan has ended.

wuzhzn commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+
+### Implementation of MedicineManager class, Medicine class
+
+![img_1.png](diagrams%2FMedicineSequenceDiagram.png)
```
Perhaps the new Medicine should include the constructor of the Medicine class and classes in the sequence diagram should include colons.

Arif-Khalid replied:
Should the line for AnalyseIllness be dotted?

wuzhzn reviewed (commented):
There is very detailed explanation and sufficient UML diagram to understand the flow of the project well, perhaps there can be some minor edits to the UML with regards to the syntax

chinhan99 commented on docs/diagrams/DiagnosisClassDiagram.png:
Incorrect syntax used. "C" logo should be removed. "+" and "-" statuses should be shown instead of squares and circles too.

chinhan99 commented on docs/diagrams/DiagnosisSequenceDiagram.png:
class boxes at the bottom should be removed

chinhan99 commented on docs/diagrams/UIClass.png:
remember to update the syntax, and also link the classes together if necessary

chinhan99 commented on docs/diagrams/loadDataSequenceDiagram.png:
The numbering should also be removed.

chinhan99 commented on docs/diagrams/saveDataSequenceDiagram.png:
"write medicine history to file" is the process of writeMedicineHistory() , so theres is no need to use self-referencing arrows in that activation.

chinhan99 reviewed (commented):
Good attempt on DG, do note the syntax errors, as mentioned during the tutorial class

Geeeetyx committed:
Update DG

Geeeetyx committed:
Update DG

Geeeetyx committed:
Merge branch 'master' into Geeeetyx_DG5

Geeeetyx committed:
Merge pull request #108 from Geeeetyx/Geeeetyx_DG5

Update DG

Zeno-Zr commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+| v1.0    | office worker feeling a bit ill     | quickly get medicine during lunch break                                                                          | get back to work quickly                                                                               |
+| v1.0    | patient who dislikes seeing doctors | be prescribed medication without consulting a doctor                                                             | avoid interacting with doctors.                                                                        |
+| v1.0    | patient                             | to know whether my symptoms are serious enough to warrant a trip to the doctor or would some medicine be enough. | save time on visiting a doctor.                                                                        |
+| v1.0    | medical professional                | identify more severe illnesses that patients might have                                                          | be referred to by Doctor Duke so that the patient can get the medical care required.                   |
```
Does this fit well to the Target user profile? This feels like an extension to your target use profile, which states it targets users with mild illnesses.

Zeno-Zr commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+`login()` is used to log a patient into the system.
+`exit()` is used to exit the program.
+`showAccountMenu()` is used to display the list of choices to the user.
+`getUserSymptoms` is to ask input form user and store it in an arraylist.
```
Minor issue here, getUserSymptoms should have brackets after it as seen in the user interface diagram. Overall, great job with the formatting of the DG, which looks very neat

Zeno-Zr commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
-{Describe the value proposition: what problem does it solve?}
+The current median wait times at a polyclinics across Singapore average at around 15-20 minutes,
+(Source: 
+[MOH](https://www.moh.gov.sg/resources-statistics/healthcare-institution-statistics/waiting-times-for-registration-and-for-consultation-at-polyclinics)).
```
It is nice that sources are given to justify any statistics quoted in the DG, which strengthens your value propositions. Great job!

Zeno-Zr commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+The Patient class contains important getters and setters, allowing different classes to access the patient object and
+the patient's information via the following methods:
+
+`getName()` returns the name of the patient.
```
Might be good to standardize how the methods are formatted.
Personally, I like the way the methods are displayed here (row by row) rather than all in a single paragraph like the rest of the DG as seen in this image
![image](https://user-images.githubusercontent.com/87852386/228461137-d3e1cfdb-bc72-4c5e-b4bf-523a22697c4d.png)

Zeno-Zr reviewed (commented):
Overall, good job with the DG. it's detailed enough to give a brief understanding of the whole program while not going too in depth which makes it easy to digest.

Thunderdragon221 committed:
correct DG diagrams

Thunderdragon221 committed:
Merge pull request #110 from Thunderdragon221/branch-correctDG

correct DG diagrams

Thunderdragon221 committed:
add JUnitTests for Storage class

Thunderdragon221 committed:
Merge pull request #112 from Thunderdragon221/branch-addJUnitTest

add JUnitTests for Storage class

Thunderdragon221 committed:
add defense for Storage class

Thunderdragon221 committed:
Merge pull request #114 from Thunderdragon221/branch-defendStorageClass

add defense for Storage class

FanZixian commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+[comment]: <> (//@@Thunderdragon221)
+### Storage
+
+![StorageClassDiagram.png](diagrams%2FStorageClassDiagram.png)
```
Is that we also need to put alter in the self invoke part as we there are lots to things to check to make sure it works

Arif-Khalid replied:
Should the inclusion of exceptions in class diagrams be made consistent across all class diagrams?

Arif-Khalid commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+continues to run until the user exits the application. The menu class allows patients to log in, register and exit the
+program. 
+
+`showWelcomeMenu()` is used to display the welcome menu to the user.
+`register()` is used to register a new patient into the system.
+`login()` is used to log a patient into the system.
+`exit()` is used to exit the program.
+`showAccountMenu()` is used to display the list of choices to the user.
+`getUserSymptoms` is to ask input form user and store it in an arraylist.
+`addSymptomToSymptomList()` is used to add the symptom to the symptom list.
+`parseSymptomInput()` is used to parse the symptom input from the user.
+`addSymptoms()` is used to add the symptoms to the symptom list.
+`displaySymptomsList()` is used to display the list of symptoms to the user.
```
Wouldn't it be clearer to show the significant methods through a sequence diagram which calls it such as one that shows the execution of the parser

Arif-Khalid replied:
Do you need to explain self-explanatory methods such as addSymptoms?

FanZixian commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+[comment]: <> (//@@author Brennanzuz)
+### Diagnosis
+
+![DiagnosisClassDiagram.png](diagrams%2FDiagnosisClassDiagram.png)
```
Maybe add more details for the aggregation as it is a little bit hard to understand.

Arif-Khalid replied:
Does this class diagram follow the UML standards for accessibility?

Arif-Khalid replied:
Do you need to include constructors in class diagram?

Arif-Khalid replied:
Should there be the C in front of the names of the classes?

FanZixian commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+Dummy accounts are provided for testing. First, access the Doctor Duke programme by following the steps in the
+User Guide, which can be found [here](https://ay2223s2-cs2113-w13-1.github.io/tp/UserGuide.html).
```
Good practice to connect DG to UG

FanZixian commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+
+![DiagnosisClassDiagram.png](diagrams%2FDiagnosisClassDiagram.png)
+
+#### Symptoms
```
Is that the all possible symptoms in enum class exactly all to patients? Maybe refer to some webs or add a enum to represent all other symptoms.

Arif-Khalid commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+file.
+
+### Implementation of loadData() in Storage Class
+![loadDataSequenceDiagram.png](diagrams%2FloadDataSequenceDiagram.png)
```
Should the lifespan of the entity continue after deletion?

Arif-Khalid commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+![loadDataSequenceDiagram.png](diagrams%2FloadDataSequenceDiagram.png)
+
+### Implementation of saveData() in Storage Class
+![saveDataSequenceDiagram.png](diagrams%2FsaveDataSequenceDiagram.png)
```
Should the lifespan of the entity continue after deletion?

darrenlsx replied:
Perhaps could give a brief explanation after each implementation

Arif-Khalid commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+[comment]: <> (//@@Geeeetyx)
+
+### Patient
+![PatientClassDiagram.png](diagrams%2FPatientClassDiagram.png)
```
Should the name of the class have the C in front?

Arif-Khalid replied:
Is this the correct accessibility modifier display?

Arif-Khalid replied:
Do class diagrams need to include constructor?

Arif-Khalid commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+
+`getName()` returns the name of the patient.
+
+`getPassword()` returns the password of the patient's account.
+
+`getPatientDiagnosisHistory()` allows for access to the array where past diagnoses are stored.
+
+`getPatientMedicineHistory()` allows for access to the array where past prescribed medications are stored.
+
+`setName()` and `setPassword()` allow the programme to set the patient's name and password.
+
+`setPatientDiagnosisHistory()` and `setPatientMedicineHistory()` allow the programme to read data from a storage file,
+and then set the stored data into the Patient object.
+
+`updatePatientMedicineHistory()` and `updatePatientDiagnosisHistory()` allow us to add new diagnoses and medications
+prescribed to the Patient object.
+
+`appendMedicineToSameDate()` appends medicine patient is prescribed if additional medicine is prescribed on the same
+day, allowing us to keep track of multiple prescriptions to one patient in one session.
```
Should there be an explanation for self-explanatory methods such as getters and setters?

darrenlsx commented on docs/DeveloperGuide.md:
```diff
@@ -2,37 +2,254 @@
+[comment]: <> (//@@Geeeetyx)
+
+### Patient
+![PatientClassDiagram.png](diagrams%2FPatientClassDiagram.png)
```
I think it is fine to include constructors as methods. Make sure no overloading though.

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
feat: update MedicineManager with medicine descriptions

feat: add a queue number system

Geeeetyx committed:
Merge pull request #117 from Geeeetyx/Geeeetyx_4

Add queue number system and update some medicine descriptions

Geeeetyx committed:
feat: update MedicineManager with more medicines

Geeeetyx committed:
Merge pull request #118 from Geeeetyx/Geeeetyx_4

feat: update MedicineManager with more medicines

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
fix: update UG

Geeeetyx committed:
Merge pull request #119 from Geeeetyx/Geeeetyx_DG6

update UG

Thunderdragon221 committed:
fixes bug in data file

Thunderdragon221 committed:
Merge pull request #159 from Thunderdragon221/branch-bugFix3

fixes bug in data file

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp into NewMaster

Jeraldchen committed:
Add Main sequence diagram, edit Class Diagrams and add Architecture Diagram

Jeraldchen committed:
Merge pull request #161 from Jeraldchen/Jeraldchen-SequenceDiagrams

Jeraldchen sequence diagrams

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Thunderdragon221 committed:
Fix bugs and defend data.

Jeraldchen committed:
Add author for reposense tracking

Thunderdragon221 committed:
Merge pull request #164 from Thunderdragon221/branch-defendData

Fix bugs and defend data.

Jeraldchen committed:
Merge pull request #165 from Jeraldchen/master

Add author for Reposense compatibility

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Thunderdragon221 committed:
Update DG for Storage class

Thunderdragon221 committed:
Merge pull request #167 from Thunderdragon221/branch-updateDG

Update DG for Storage class

tanyizhe committed:
Fix typos in UG

Updated typos in UG to correctly match current application.

tanyizhe committed:
Update UserGuide.md

Fixed typos in UG

tanyizhe committed:
Update UserGuide.md

Fixed typo.

Thunderdragon221 committed:
Exit program upon finding corruption

Thunderdragon221 committed:
Merge pull request #169 from Thunderdragon221/branch-exitUponFindingCorruption

Exit program upon finding corruption

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Geeeetyx committed:
fix: fixed issue of Dr Duke diagnosing when a patient does not enter any symptoms

fix: update delete symptom feature to re-diagnose patient after deleting a symptom

Geeeetyx committed:
fix: update code style

Geeeetyx committed:
Merge pull request #170 from Geeeetyx/Geeeetyx_6

Update deleting symptom and other bugs

Jeraldchen committed:
Added PPP draft

Jeraldchen committed:
Merge pull request #171 from Jeraldchen/Jeraldchen-PPP

Jeraldchen ppp

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Commit:
Updated Developer guide
Created more MedicineManager Junit tests

Thunderdragon221 committed:
Show date of diagnosis

Commit:
Fix coding standard violation for import statement.

tanyizhe committed:
Merge pull request #172 from tanyizhe/master

Updated Developer guide and tests

tanyizhe committed:
Update DeveloperGuide.md

updated DG acknowledgements.

Thunderdragon221 committed:
Merge branch 'master' into branch-addDiagnosisDate

Thunderdragon221 committed:
Merge pull request #173 from Thunderdragon221/branch-addDiagnosisDate

Show date of diagnosis

Thunderdragon221 committed:
Resolve code functionality issues

Thunderdragon221 committed:
Resolve code functionality issues

Thunderdragon221 committed:
Merge pull request #174 from Thunderdragon221/branch-resolveMergeConflict

Resolve code functionality issues

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Brennanzuz committed:
Fixed diagram formatting

Added brennanzuz.md which details brennanzuz's contributions as required by the module.
Fixed formatting errors in all diagrams including icons in class diagrams and foot boxes in sequence diagrams.

Brennanzuz committed:
Merge pull request #175 from Brennanzuz/branch-brannon-202303071516

Fixed diagram formatting

Thunderdragon221 committed:
Update UG with corruption info.

Thunderdragon221 committed:
Merge pull request #176 from Thunderdragon221/branch-updateUG

Update UG with corruption info

Brennanzuz committed:
Added link to brennanzuz.md

Fixed missing link towards PPP.

Brennanzuz committed:
Merge pull request #177 from Brennanzuz/branch-brannon-202303071516

Added link to brennanzuz.md

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Jeraldchen committed:
Fix invalid symptom choice message to make it clearer for users

Jeraldchen committed:
Merge pull request #179 from Jeraldchen/Jeraldchen-FixInvalidCommandBug

Jeraldchen fix invalid command bug

Jeraldchen committed:
Edit link to PPP

Jeraldchen committed:
Merge pull request #180 from Jeraldchen/Jeraldchen-FixInvalidCommandBug

Edit link to PPP

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Brennanzuz committed:
Added SymptomHandler.java

Added SymptomHandler.java to deal with conversion of symptoms from their enumeration values to readable strings.
Changed the way symptoms are printed out when listing symptoms (command 4) and deleting symptoms (command 5).

Jeraldchen committed:
Add trim() to remove whitespaces for choice selection input

Jeraldchen committed:
Merge pull request #181 from Jeraldchen/Jeraldchen-WhitespaceBug

Jeraldchen whitespace bug

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Brennanzuz committed:
Changed illness match decimal points

Changed number of decimal points displayed for each illness candidate's matching percentage to always be 2.

Brennanzuz committed:
Merge branch 'master' into branch-brannon-202303071516

Brennanzuz committed:
Merge pull request #182 from Brennanzuz/branch-brannon-202303071516

Branch brannon 202303071516

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Allow user to go back to symptom menu after typo input

Jeraldchen committed:
Merge pull request #183 from Jeraldchen/Jeraldchen-StayInSameMenu

Jeraldchen stay in same menu

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Edit reposense link

Jeraldchen committed:
Merge pull request #184 from Jeraldchen/master

Edit Reposense link in PPP

Thunderdragon221 committed:
Add PPP and resolve bugs.

Thunderdragon221 committed:
Merge pull request #185 from Thunderdragon221/branch-addPPP

Add PPP and resolve bugs.

Commit:
Improve delete function.
Added PPP for tanyizhe.

Commit:
improve Delete symptom function.
Added PPP draft.

Commit:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	docs/AboutUs.md

Commit:
Updated UG for deleting symptom feature.

Commit:
Fix coding standard violation.

tanyizhe committed:
Merge pull request #186 from tanyizhe/master

Improve delete symptom feature. Added PPP draft.

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
Update DG with PPP

Geeeetyx committed:
Merge pull request #187 from Geeeetyx/Geeeetyx_DG7

Update DG with PPP

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
feat: introduce more OTC medications

Geeeetyx committed:
Merge pull request #188 from Geeeetyx/Geeeetyx_7

Introduce more medicines

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Brennanzuz committed:
Changed symptom name display

Added new JUnit test SymptomTest.java to test for output of SymptomHandler.toString().
Added SymptomHandler.printSymptoms() to print all symptoms given an array of them.
Changed way to print symptom to use SymptomHandler.printSymptoms() in Information.java and Menu.java.
Changed link in AboutUs.md to html.
Fixed format in DiagnosisTest.java.

Brennanzuz committed:
Merge pull request #189 from Brennanzuz/branch-brannon-202303071516

Changed symptom name display

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Add more J unit tests

Jeraldchen committed:
Edit spelling error in Junit Test

Jeraldchen committed:
Edit spelling error

Jeraldchen committed:
Merge pull request #193 from Jeraldchen/Jeraldchen-IncreaseTestCoverage

Jeraldchen increase test coverage

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Brennanzuz committed:
Fixed outdated output when displaying symptoms in UserGuide.md

Changed text displaying symptoms in the UserGuide.md after it has been changed to lowercase.

Added note under login in UserGuide.md to explain why having multiple passwords for the same username to access different accounts is intentional.

Added note under "reporting symptoms" in UserGuide.md in the case that the patient's symptoms does not yield any suitable diagnosis.

Added separating line in Information.java when viewing symptom history in viewSymptomHistory().

Tried to fix image not showing in brennanzuz.md.

Brennanzuz committed:
Merge pull request #195 from Brennanzuz/branch-brannon-202303071516

Fixed outdated output when displaying symptoms in UserGuide.md

Brennanzuz committed:
Tried to fix image not showing in brennanzuz.md.

Brennanzuz committed:
Merge pull request #196 from Brennanzuz/branch-brannon-202303071516

Tried to fix image not showing in brennanzuz.md.

Brennanzuz committed:
Tried to fix image not showing in brennanzuz.md.

Brennanzuz committed:
Merge pull request #197 from Brennanzuz/branch-brannon-202303071516

Tried to fix image not showing in brennanzuz.md.

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Remove typo in PPP

Jeraldchen committed:
Merge pull request #198 from Jeraldchen/master

Remove typo in PPP

Jeraldchen committed:
Update project status on PPP

Jeraldchen committed:
Merge pull request #199 from Jeraldchen/master

Update project status on PPP

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Geeeetyx committed:
Merge branch 'master' of https://github.com/Geeeetyx/tp

Geeeetyx committed:
fix: update patient class diagram, update about us

Geeeetyx committed:
Merge pull request #200 from Geeeetyx/Geeeetyx_DG8

Update about us

Commit:
Finalise PPP

tanyizhe committed:
Merge pull request #201 from tanyizhe/tanyizhe-PPP

Finalise PPP

tanyizhe committed:
Update tanyizhe.md

Formatted PPP

tanyizhe committed:
Update tanyizhe.md

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Edit sequence diagram arrows

Jeraldchen committed:
Replace sequence diagram image

Jeraldchen committed:
Merge pull request #203 from Jeraldchen/Jeraldchen-EditDG

Jeraldchen edit dg

Jeraldchen committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W13-1/tp

Jeraldchen committed:
Update JAR file name in UG

Jeraldchen committed:
Edit JAR file name in UG

Jeraldchen committed:
Merge pull request #204 from Jeraldchen/master

Edit JAR file name in UG

Jeraldchen committed:
Update UserGuide.md

Jeraldchen committed:
Update UserGuide.md
